sys/net/application_layer/dhcpv6/client.c

```diff
@@ -802,6 +802,9 @@ static void _parse_advertise(uint8_t *adv, size_t len)
          len > 0; len -= _opt_len(opt), opt = _opt_next(opt)) {
         switch (byteorder_ntohs(opt->type)) {
             case DHCPV6_OPT_IA_PD:
+                if (_opt_len(opt) < sizeof(dhcpv6_opt_ia_pd_t)) {
+                    return;
```

Comment:
Maybe, before each return, a `DEBUG` message could be helpful, tbh.

Reply:
Done